### PR TITLE
Flex search internal db - starts_with

### PR DESCRIPTION
**Pull request by mfusser**
starts_with is required to test the search field

### CODE REVIEW

File: src/database/inmemory/js-generator.ts [hunk 1]
```diff
@@ -859,6 +860,12 @@ register(FieldPathQueryNode, (node, context) => {
     return fragment;
 });
 
+register(FlexSearchStartsWithQueryNode, (node, context) => {
+    const lhs = processNode(node.lhs, context);
+    const rhs = processNode(node.rhs, context);
+    return js`(${lhs}.startsWith(${rhs}))`;
```
Comment (Yogu) [inline src/database/inmemory/js-generator.ts:866]: this is missing a null check on `lhs`. In `BinaryOperationQueryNode`, I used js`String(${lhs}).startsWith(${rhs})`. You could add a test where you use `_starts_with` on a field that has null values somewhere (e.g. `Country.totalInvestment`).

Comment (mfusser) [reply src/database/inmemory/js-generator.ts:866]: Country.totalInvestment doesn't work, because of permissions. I will add a new field to delivery that is null in some cases.
